### ISSUE TITLE
As a data provider, I want to be able to provide date/time stamps with >6 decimal places of precision

### ISSUE THREAD

**Issue by cgobat** [NASA-PDS/pds4-information-model#672]
### Checked for duplicates

Yes - I've already checked

### 🧑‍🔬 User Persona(s)

Any data providers or users who deal with/need/have times at finer-than-millisecond precision

### 💪 Motivation

...so that I don't have to discard/truncate the perfectly valid time precision that I have available when creating data labels.

### 📖 Additional Details

For IM versions <1K00, the `ASCII_Date_Time_*` classes are governed by a formation rule/regular expression pattern where the fractional seconds are given by the group `(\.([0-9]{1,6}))?`, which denotes an optional field consisting of a `.` followed by between 1 and 6 digits (characters 0 through 9). [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) doesn't define any upper limit to the number of digits of fractional second precision. Simply changing the `{1,6}` in these patterns to `+` (which would make it `(\.([0-9]+))?`) would resolve this.

### Acceptance Criteria

**Given** 
**When I perform** 
**Then I expect**

**Comment by radiosci** [comment]
The number of digits in fractional seconds   fff   should not be restricted so long as it is commensurate with precision (see Standards Reference 5A.2, top of page 32 in the current version).  This question has been raised before (see CCB-226 in JIRA); if validation is returning an error when   fff   has more than 6 digits, a bug report should be submitted.

**Comment by cgobat** [comment]
I agree that it *should* not be restricted, but it is.[^vnote] Take a look at the relevant regular expression pattern for `ASCII_Date_Time_YMD_UTC` (from [PDS4_PDS_1J00.xsd](https://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1J00.xsd)), for example:
```xml
<xs:pattern value="(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)(([0-1][0-9])|(2[0-3])):[0-5][0-9]:(([0-5][0-9])|60)(\.([0-9]{1,6}))?(Z)"/>
```

The group immediately preceding the terminating "Z" places a limit of 6 digits on that field. Same goes for `ASCII_Date_Time_DOY`, `ASCII_Date_Time_DOY_UTC`, and `ASCII_Date_Time_YMD`. As it turns out, I just noticed that the analogous patterns for `ASCII_Date_Time` and `ASCII_Date_Time_UTC` themselves actually only allow a maximum of 4 decimal digits. Here is an example of the validation failure you get if you have an `ASCII_Date_Time_YMD_UTC` with >6 digits after the decimal:
```
cvc-pattern-valid: Value '2000-01-01T12:00:00.0000000Z' is not facet-valid with respect to pattern '(-)?[0-9]{4}(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)(([0-1][0-9])|(2[0-3]))(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)(([0-1][0-9])|(2[0-3])):[0-5][0-9](Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)(([0-1][0-9])|(2[0-3])):[0-5][0-9]:(([0-5][0-9])|60)(\.([0-9]{1,6}))?(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)24((:00)|(:00:00))?(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(T)24:00:00(\.([0]{1,6}))(Z)|(-)?[0-9]{4}-((0[1-9])|(1[0-2]))-((0[1-9])|([1-2][0-9])|(3[0-1]))(Z)' for type 'ASCII_Date_Time_YMD_UTC'.
```

---

Based on my read of it, CCB-226 dealt with allowing microsecond precision (i.e., *up to* 6 decimal digits). This issue has to do with being *limited to* 6.

[^vnote]: In IM versions &le;1J00, that is. I do note that this is no longer a problem in 1.20.0.0 (1K00), as the decimal piece of the pattern has been updated to `(\.([0-9]+))?`, as I suggested in the original text of this issue. I guess if that is as good as it gets, and there is no expectation that such fixes will be backported, this issue can be closed as not planned.

**Comment by radiosci** [comment]
The limitation to 6 decimal digits, among other things, was revisited in CCB-350 (attached), which should have been implemented in IM 1.20.0.0 (1K00) in June 2023.
[CCB-350-2306xx.pdf](https://github.com/NASA-PDS/pds4-information-model/files/12593901/CCB-350-2306xx.pdf)

**Comment by jordanpadams** [comment]
Per https://github.com/NASA-PDS/pds4-information-model/issues/672#issuecomment-1716999413, closing this as wontfix
